apps/web/components/core/application-header.tsx

```diff
@@ -1,17 +1,115 @@
-"use client";
-
-import { AppShell, Burger, Group } from "@mantine/core";
+import { AppShell, Avatar, Burger, Button, Group, Menu } from "@mantine/core";
+import { useSupabaseStore } from "@stores/supabase-store";
+import type { Session } from "@supabase/auth-helpers-nextjs";
 import Link from "next/link";
+import { useEffect, useState } from "react";
+
+interface ApplicationHeaderProps {
+	opened: boolean;
+	toggle: () => void;
+}
+
+// Assuming proper typing for SupabaseClient is completed outside this snippet
+export default function ApplicationHeader({ opened, toggle }: ApplicationHeaderProps): JSX.Element {
+	const [sessionData, setSessionData] = useState<Session | null>(null);
+	const { supabase, isLoggedIn } = useSupabaseStore();
+
+	const [working, setWorking] = useState(false);
+
+	useEffect(() => {
+		const fetchSession = async (): Promise<void> => {
+			if (!supabase?.auth) return;
+
+			try {
+				const {
+					data: { session }
+				} = await supabase.auth.getSession();
+				setSessionData(session);
+			} catch (error) {
+				console.error(error);
+			}
+		};
+
+		fetchSession().catch(console.error);
+	}, [supabase]);
+
+	const signOut = async (): Promise<void> => {
+		if (!supabase?.auth) return;
+
+		try {
+			await supabase.auth.signOut();
+		} catch (error) {
+			console.error(error);
+		}
+	};
+
+	const signInWithAzure = async (): Promise<void> => {
+		if (!supabase?.auth) return;
+
+		try {
+			const result = await supabase.auth.signInWithOAuth({ provider: "azure" });
```

Comment:
Also can you add this so it redirects to the same page after authentication? (right now it always goes to the first page)

```ts
supabase.auth.signInWithOAuth({
	provider: "azure",
	options: { redirectTo: window.location.href }
})
```